ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixAggregatedViewClusterInitializer.java

```diff
@@ -179,6 +176,36 @@ helixClusterManager.new HelixClusterChangeHandler(clusterMapConfig.clusterMapClu
       clusterChangeHandler.waitForInitNotification();
     }
 
+    // 3. Fetch live instance changes in entire cluster
+    // When adding a listener, current live instances are fetched from Helix using blocking call and notified to the
+    // listener (Please follow below method to
+    // https://github.com/apache/helix/blob/fb773839117ee0689451fab7c8b949115f555100/helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java#L393).
+    // I.e. by the time we return from below method, initial notification of live instances states would have been
+    // arrived. So, we don't need separate latch to wait for notification of live instances.
+    helixManager.addLiveInstanceChangeListener(clusterChangeHandler);
+    logger.info("Registered live instance change listeners for entire cluster via Helix manager at {}",
+        localZkConnectStr);
+
     return new HelixAggregatedViewClusterInfo(helixManager, clusterChangeHandler, dataNodeConfigSources);
   }
+
+  /**
+   * Register for data node configs from Helix Property store.
+   * @param dcName data center for which we are interested in data node configs.
+   */
+  private void registerForDataNodeConfigsChanges(String dcName) throws Exception {
+    DcZkInfo dcZkInfo = dataCenterToZkAddress.get(dcName);
+    String zkConnectStr = dcZkInfo.getZkConnectStrs().get(0);
+    DataNodeConfigSource dataNodeConfigSource =
+        helixFactory.getDataNodeConfigSource(clusterMapConfig, zkConnectStr, dataNodeConfigSourceMetrics);
+    // When attaching a listener, current data is fetched from Helix using blocking call and notified to the listener
+    // (Please see PropertyStoreToDataNodeConfigAdapter.Subscription#start() method). I.e. by the time we return from
+    // below method, initial notification of data node configs would have been arrived. So, we don't need separate latch
+    // to wait for notification of data node configs.
+    dataNodeConfigSource.addDataNodeConfigChangeListener(
```

Comment:
Yeah, I was thinking that since the datacenter name returned in the DataNodeConfig is updated by bootstrap tool and could be incorrect, thought of relying on the data center name passed in the caller function.